enhancements/ignition-spec-dual-support.md

```diff
@@ -0,0 +1,447 @@
+installer and underlying control plane/bootstrap configuration, as well as RHEL 
+CoreOS (Henceforth RHCOS) package version will also be updated.
+
+This overall migration will take part as a two-phase process:
+
+Phase 1/OCP 4.4:
+The objective of this phase is to allow users to apply machineconfigs with
+ignition spec 3 support to new and existing clusters. The breakdown is:
+
+ - a tool is created to translate between ignition spec versions
+ - MCD gains the ability to process both spec 2 and spec 3 configs
+ - MCC gains the ability to translate from spec 3 configs to spec 2 configs
+
+Some other work that would be nice to have in phase 1 to prepare for phase 2,
+but not necessary to acheive the objective:
+
+ - MCC attempts to translate all machineconfigs with v2 ignition config to v3
+ - MCC learns to generate v3 initial master/worker ignition configs
+
+
+Phase 2/OCP 4.X:
+All new installs will be on spec 3 only. Existing clusters that have non spec 3
+ignition config machineconfigs will not be allowed to update to the new version.
+
+ - RHCOS bootimages switches to only accept ignition spec 3 configs
+ - The Openshift installer is updated to generate spec 3 configs
+ - Remaining MC* components generate spec 3 only
+ - MCO enforces that all configs are spec 3 before allowing the CVO to start the update
+
+
+## Motivation
+
+Ignition released v2.0.0 (stable) on Jun 3rd, 2019, which has an updated
+specification format (ignition spec version 3, henceforth “Spec 3”). This
+change includes some important reworks for RHCOS provisioning, most
+importantly the ability to specify and mount other filesystems and fixing
+issues where ignition v0 spec was not declarative. Particularly, this is
+required to support having /var on a separate partition or disk, an important
+requirement for security/compliance purposes in OCP. The existing version on
+RHCOS systems (ignition version v0.33) carries a spec version (spec version 2,
+henceforth “Spec 2”) that is not compatible with Spec 3. Thus we would like to
+update the ignition version on RHCOS/Installer/MCO to make use of the changes.
+
+This proposal will also allow closer alignment with OKD, as OKD will be based
+on Fedora CoreOS (Henceforth FCoS), which is already on, and only supports,
+ignition spec 3. We want to do this in a way that can minimize deltas between
+OKD and OCP.
+
+
+### Goals
+
+#### Phase 1:
+- [ ] A config translator is created to translate from ignition spec 2 to spec 3
+- [ ] A config translator is created to translate from ignition spec 3 to spec 2
+- [ ] The MCO is updated to support both ignition spec 2 and 3, with:
+    - [ ] MCC rendering spec 3 configs to spec 2
+    - [ ] MCD gaining the ability to process both spec 2 and spec 3 configs
+- [ ] Create tests to apply spec 3 configs to spec 2 clusters successfully
+- [ ] An alerting mechanism is put in place for outdated and incompatible/non-translatable configs
+- [ ] Docs are updated to reflect the new config version
```

Comment:
@crawford can you ack this plan if it does look good to you? Not sure who else to bring in